Assets/Scripts/Movement.cs

```diff
@@ -7,20 +7,22 @@ public class Movement : MonoBehaviour
 {
     private CharacterController characterController;
     public Vector2 movementVector;
+
     void Start()
     {
         characterController = GetComponent<CharacterController>();
     }
+
     public void PlayerMovement(InputAction.CallbackContext context)
     {
         movementVector = context.ReadValue<Vector2>();
-        if (context.started)
-        {
-            Debug.Log("KLIK!");
-        }
-        if (context.canceled)
+    }
+
+    private void Update()
+    {
+        if (movementVector != new Vector2(0, 0)) 
```

Comment:
Niepotrzebnie - jeśli odczytany wektor jest zerem, to śmiało możemy "ruszać" postać o 0 :) 
Tutaj tworzymy co klatkę nową zmienną, co zajmuje trochę czasu! W takim przypadku lepiej użyć stałej (np Vector2.zero).
Jeśli chcesz wprowadzić tutaj "dead-zone" dla kontrolerów, to możesz wykonywać ruch, tylko gdy długość wektora ruchu (magnitude) jest większa od pewnej stałej (np: movementVector.magnitude >= deadZone).

Reply:
No ale w takim razie i tak się to musi znajdować w funkcji Update tak?

Reply:
Tak

Reply:
Albo i nie - możemy działać na eventach